src/Microsoft.Extensions.Logging.Filter/IFilterLoggerSettings.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Filter settings for messages logged by an <see cref="ILogger"/>.
+    /// </summary>
+    public interface IFilterLoggerSettings
```

Comment:
Being on the interface is probably okay - it's following the same pattern that the console's logger's build-in filtering+reloadablefile featues. Kicker is that the filter needs to know how to react to variable switches without "asking" for the switch literally every time and without being tightly coupled to a specific switch provider.

I guess this could be split into two interfaces, but that seems kind of excessive for fairly buried interface which site owners won't typically implement.